Signal/src/AppDelegate.m

```diff
@@ -44,23 +45,39 @@ @implementation AppDelegate
 #pragma mark Detect updates - perform migrations
 
 - (void)performUpdateCheck{
-    // We check if NSUserDefaults key for version exists.
     NSString *previousVersion = Environment.preferences.lastRanVersion;
-    NSString *currentVersion  = [Environment.preferences setAndGetCurrentVersion];
-    
+    NSString *currentVersion = [Environment.preferences setAndGetCurrentVersion];
     if (!previousVersion) {
         DDLogError(@"No previous version found. Possibly first launch since install.");
         [Environment resetAppData]; // We clean previous keychain entries in case their are some entries remaining.
-    } else if ([currentVersion compare:previousVersion options:NSNumericSearch] == NSOrderedDescending){
-        [Environment resetAppData];
-        // Application was updated, let's see if we have a migration scheme for it.
-        if ([previousVersion isEqualToString:@"1.0.2"]) {
-            // Migrate from custom preferences to NSUserDefaults
-            [VersionMigrations migrationFrom1Dot0Dot2toLarger];
+    }
+    else if ([Environment.preferences getIsMigratingToVersion2Dot0] || [currentVersion compare:previousVersion options:NSNumericSearch] == NSOrderedDescending){
+        if([self isVersion:previousVersion atLeast:@"1.0.2"]) {
+            [Environment.preferences setIsMigratingToVersion2Dot0:YES];
+            [VersionMigrations migrateFrom1Dot0Dot2ToVersion2Dot0]; // this is only necessary for older apps
```

Comment:
I left this in, although it seems unneeded for 1.0.8, I'm not sure which version it is needed for (if it still is) but as it doesn't have negative side effects left in case. but can excise if necessary